### PR TITLE
Practitioner supplychain report by practitioner

### PR DESCRIPTION
**Pull request by maimoonak**
Fixes [link to issue]

**Engineer Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [ ] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

**Review by pld — DISMISSED** [review]
@AnnieMungai @dubdabasoduba not we're not prebuilding the practioner based reports so they will be slower to load, we can add that as a later feature, but as long as we're managing expections on practitioner based report build time I think this is ok as is for now

**Review by pld — COMMENTED** [review]
@maimoonak this looks great, can you please add coverage for the new lines and take a look at this failing test, `MeasureReportRepositoryTest.testEvaluatePopulationMeasureHandlesBadMeasureUrl`

File: android/engine/src/main/java/org/smartregister/fhircore/engine/util/ReflectionUtil.kt [hunk 1]
```diff
@@ -46,3 +48,22 @@ suspend fun <T : Any> KClass<T>.callSuspendFunctionOnField(
       .apply { isAccessible = true }
   return declaredObjectMethod.callSuspend(declaredObject, *args)
 }
+
+inline fun <reified T : Any> getPrivateProperty(property: String, obj: T): Any? {
```
Comment (ellykits) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/util/ReflectionUtil.kt:52]: Was this implemented as workaround? Reflection code is less performant and would require code refactoring on API changes.

**Review by ellykits — CHANGES_REQUESTED** [review]

**Comment by pld** [comment]
@maimoonak is this a change to the upstream libraries that would be generally useful? If so, let's create an issue in the library that holds `org.opencds.cqf.cql.evaluator.measure.r4.R4MeasureProcessor` suggesting the changes we have here

**Comment by maimoonak** [comment]
The PR had a fix for practitioner reports filter library parameter which has been merged in 0.2.5 but in main it can not be merged, the reason is main and this PR uses an old version of sdk-workflow which is rightnow incompatible with new version of sdk-engine.
We are using an old commit (595f422af2723b1993929965ad876172850d8f0c) to test and enhance reports on main rightnow and current main can not run measure reports due issues in sdk versions...
So effectively the PR is useless for main unless we have updated workflow library, and can be considered as blocked until we can make main stable with engine, data-capture, and workflow libs from latest sdk. The PR is not going to help with any working in main and should not be merged to avoid needless changes

**Comment by pld** [comment]
Closing since we'll need to reimplement as @maimoonak noted

**Pull request closed without merging**